commit de7e3f88dd5c594d9ea1fac6a176716435d905bc
Author: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
Date:   Wed Oct 17 10:15:34 2018 +0200

    staging: rtl8723bs: Fix the return value in case of error in 'rtw_wx_read32()'
    
    [ Upstream commit c3e43d8b958bd6849817393483e805d8638a8ab7 ]
    
    We return 0 unconditionally in 'rtw_wx_read32()'.
    However, 'ret' is set to some error codes in several error handling paths.
    
    Return 'ret' instead to propagate the error code.
    
    Fixes: 554c0a3abf216 ("staging: Add rtl8723bs sdio wifi driver")
    Signed-off-by: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/staging/rtl8723bs/os_dep/ioctl_linux.c b/drivers/staging/rtl8723bs/os_dep/ioctl_linux.c
index c38298d960ff..4f120e72c7d2 100644
--- a/drivers/staging/rtl8723bs/os_dep/ioctl_linux.c
+++ b/drivers/staging/rtl8723bs/os_dep/ioctl_linux.c
@@ -2289,7 +2289,7 @@ static int rtw_wx_read32(struct net_device *dev,
 exit:
 	kfree(ptmp);
 
-	return 0;
+	return ret;
 }
 
 static int rtw_wx_write32(struct net_device *dev,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1025_de7e3f88dd5c594d9ea1fac6a176716435d905bc.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1025_de7e3f88dd5c594d9ea1fac6a176716435d905bc.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
